File: Python/Dosen.py
from SivitasAkademik import SivitasAkademik

# Dosen adalah anak dari sivitas akademik
class Dosen(SivitasAkademik) :
    # Konstruktor
    def __init__(self, nik = "", nama = "", jenis_kelamin = "", asalUniv ="", emailEdu = "", nip = "", fakultas = "", pendidikanAkhir = "", keahlian = ""):
        # Mengambil data ortunya
        super().__init__(nik, nama, jenis_kelamin, asalUniv, emailEdu)
        # Inisiaslisasi untuk dirinya sendiri
        self.__nip = nip
        self.__fakultas = fakultas
        self.__pendidikanAkhir = pendidikanAkhir 
        self.__keahlian = keahlian
    
    # Get
    def get_nip(self):
        return self.__nip
    def get_fakultas(self):
        return self.__fakultas
    def get_pendidikanAkhir(self):
        return self.__pendidikanAkhir
    def get_keahlian(self):
        return self.__keahlian
    
    # Set
    def set_nip(self, nip):
        self.__nip = nip
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
    def set_pendidikanAkhir(self, pendidikanAkhir):
        self.__pendidikanAkhir = pendidikanAkhir
    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

File: C++/Dosen.cpp
using namespace std;

//Mahasiswa adalah child, sivitasAkademik adalah parent
class Dosen : public SivitasAkademik{
    // variabel yang dimiliki kelas ini
    protected:
        string nip;
        string fakultas;
        string pendidikanAkhir;
        string keahlian;

    public:
        //konstruktor
        // Dosen(){}
        // konstruktor sekaligus mengakses variabel parentnya
        Dosen(string nik,string nama, string jenis_kelamin,string asal_universitas, string email_edu, string nip, string fakultas, string pendidikanAkhir, string keahlian) : SivitasAkademik(nama, nik, jenis_kelamin, asal_universitas, email_edu){
            this->nip = nip;
            this->keahlian = keahlian;
            this->fakultas = fakultas;
            this->pendidikanAkhir = pendidikanAkhir;
        }

        // Getter dan setter
        // Getter
        string getNip() {
            return this->nip;
        }
        string getFakultas() {
            return this->fakultas;
        }
        string getKeahlian() {
            return this->keahlian;
        }
        string getPendidikanAkhir() {
            return this->pendidikanAkhir;
        }

        // Setter
        void setNip(string nip) {
            this->nip = nip;
        }
        void setFakultas(string fakultas) {
            this->fakultas = fakultas;
        }
        void setKeahlian(string keahlian) {
            this->keahlian = keahlian;
        }
        void setPendidikanAkhir(string pendidikanAkhir) {
            this->pendidikanAkhir = pendidikanAkhir;
        }

        // Destruktor
        ~Dosen(){}
};